### PR TITLE
ref: fix pyright for sentry.api.validators.monitor

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: pyrightconfig-commithook.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "include": [
     "src"
   ],
-
+  "extraPaths": ["src"],
```
Comment (asottile-sentry) [inline pyrightconfig-commithook.json:5]: this helps pyright understand our import root, fixes errors like:

```
  /Users/asottile/workspace/sentry/src/sentry/utils/email/list_resolver.py:10:28 - error: Type argument for "Generic" must be a type variable
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -59,8 +59,9 @@ repos:
       name: pyright
       entry: pyright
       language: node
+      files: ^src/
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:62]: skips `tests` and `examples`

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -59,8 +59,9 @@ repos:
...
       types: [ python ]
-      pass_filenames: true
+      require_serial: true
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:64]: `pass_filenames` is the default, pyright uses multiprocessing so `require_serial` prevents a forkbomb

Comment (wedamija) [reply .pre-commit-config.yaml:64]: Does this potentially cause pyright to take longer to run?

**Review by wedamija — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 3 above)
Comment (asottile-sentry) [reply .pre-commit-config.yaml:64]: it's ~slightly faster now:
- before there was resource contention due to pre-commit's multiprocessing + pyright's multiprocessing leading to exhaustion
- after pyright is now properly following imports so it does a little more work

`--all-files` goes from 12.7s to 12.6s in my tests

**Pull request merged**